src/actions/actions.ts

```diff
@@ -4980,7 +4990,35 @@ class ActionDeleteLineVisualMode extends BaseCommand {
   keys = ["X"];
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
-    return await new DeleteOperator(this.multicursorIndex).run(vimState, position.getLineBegin(), position.getLineEnd());
+    if (vimState.currentMode === ModeName.Visual) {
+      return await new DeleteOperator(this.multicursorIndex).run(vimState,
+                                                                vimState.lastVisualSelectionStart.getLineBegin(),
```

Comment:
Which properties you want to use?
I can find a current selection member in `vimState`.

Using `TextEditor.getSelection() `does not work.